analysis_options.yaml

```diff
@@ -1,179 +1,11 @@
-    # - one_member_abstracts # too many false positives
-    # - only_throw_errors # https://github.com/flutter/flutter/issues/5792
-    - overridden_fields
-    - package_api_docs
-    - package_names
-    - package_prefixed_library_names
-    # - parameter_assignments # we do this commonly
-    - prefer_adjacent_string_concatenation
-    - prefer_asserts_in_initializer_lists
-    - prefer_collection_literals
-    - prefer_conditional_assignment
-    - prefer_const_constructors
-    - prefer_const_constructors_in_immutables
-    - prefer_const_declarations
-    - prefer_const_literals_to_create_immutables
-    # - prefer_constructors_over_static_methods # not yet tested
-    - prefer_contains
-    - prefer_equal_for_default_values
-    # - prefer_expression_function_bodies # conflicts with https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using--for-short-functions-and-methods
-    - prefer_final_fields
-    - prefer_final_locals
-    - prefer_foreach
-    # - prefer_function_declarations_over_variables # not yet tested
-    - prefer_generic_function_type_aliases
-    - prefer_initializing_formals
-    # - prefer_int_literals # not yet tested
-    # - prefer_interpolation_to_compose_strings # not yet tested
-    - prefer_is_empty
-    - prefer_is_not_empty
-    - prefer_iterable_whereType
-    # - prefer_mixin # https://github.com/dart-lang/language/issues/32
-    - prefer_single_quotes
-    - prefer_typing_uninitialized_variables
-    - prefer_void_to_null
-    # - public_member_api_docs # enabled on a case-by-case basis; see e.g. packages/analysis_options.yaml
-    - recursive_getters
-    - slash_for_doc_comments
-    - sort_constructors_first
-    - sort_pub_dependencies
-    - sort_unnamed_constructors_first
-    - super_goes_last
-    - test_types_in_equals
-    - throw_in_finally
-    # - type_annotate_public_apis # subset of always_specify_types
-    - type_init_formals
-    # - unawaited_futures # too many false positives
-    # - unnecessary_await_in_return # not yet tested
-    - unnecessary_brace_in_string_interps
-    - unnecessary_const
-    - unnecessary_getters_setters
-    # - unnecessary_lambdas # has false positives: https://github.com/dart-lang/linter/issues/498
-    - unnecessary_new
-    - unnecessary_null_aware_assignments
-    - unnecessary_null_in_if_null_operators
-    - unnecessary_overrides
-    - unnecessary_parenthesis
-    - unnecessary_statements
-    - unnecessary_this
-    - unrelated_type_equality_checks
-    # - use_function_type_syntax_for_parameters # not yet tested
-    - use_rethrow_when_possible
-    # - use_setters_to_change_properties # not yet tested
-    # - use_string_buffers # has false positives: https://github.com/dart-lang/sdk/issues/34182
-    # - use_to_and_as_if_applicable # has false positives, so we prefer to catch this by code-review
-    - valid_regexps
-    # - void_checks # not yet tested
+    - 'lib/src/http/**'
```

Comment:
nit: add a new line at the end of file

Reply:
thanks.  I'm actually looking at moving the analysis_options_user.yaml to this file as well.  Do you know why `close_sinks` is enabled for the user version but not the main one?  Seems like we might want to disable it for both?

Reply:
>Do you know why close_sinks is enabled for the user version but not the main one?

No.

`analysis_options_user.yaml` has a lot less rules enabled.

Reply:
I'm disabling it for the `analysis_options_user.yaml` to be consistent